### DISCUSSION TITLE
next.js + bigcommerce/storefront-data-hooks + /{country}/{lang} routes on Vercel

### DISCUSSION THREAD

**Discussion by dy-byron** [vercel/next.js#25215] [Help]
We have a client that has 3 different stores on bigcommerce to sell in 3 different countries. Each country has 2 languages. The client wants to use only one domain for all routes and stores. We want to deploy it to Vercel so we can use all the features that next.js provides to us along with bigcommerce/storefront-data-hooks to communicate with each store. 

Let's say these are the countries and the language for each:
```
ES <- default country
 - es
 - it
UK
 - en
 - it
FR
 - fr
 - it
```

So the final website should like this (home page as an example):
```
client.eu/ <- ES country & es locale
client.eu/it <- ES country & it locale

client.eu/uk <- UK country & uk locale
client.eu/uk/it <- UK country & it locale

client.eu/fr <- FR country & fr locale
client.eu/fr/it <- FR country & it locale
```

Does anyone know how to accomplish this easily? 
I've few ideas but all of them break the defaults of next.js or vercel or bigcommerce/storefront-data-hooks and it does feel like overkill the project. 

Thanks!

**Comment by juliuscc** [comment]
We also have this problem. We would like to use the built-in internalization in Nextjs but that would prohibit us from having the routing that we want. We want to do: `domain.com/country-locale` or `domain.com/se-en`. The country will be the basis for what content we give our users and the language is just translations of our copy.

This is how Netflix and Stripe handle routing so we thought we chose a mature strategy that would have wide support 😅 

We have started considering how difficult it would be to build all routing logic ourselves but that is not a route we want to take.

**Comment by juliuscc** [comment]
Just to make it easier for future developers that end up here, I think this discussion is similar to [this](https://github.com/vercel/next.js/discussions/21871) discussion.

**Comment by dy-byron** [comment]
FYI, we end up using a subdomain strategy for the regions and the nextjs's built-in i18n for the locale.

es.client.eu/{locale}
fr.client.eu/{locale}
uk.client.eu/{locale}

Each one is a different project on vercel so we can assign different env variables to make storefront-data-hooks worked properly
